pghoard/pghoard.py

```diff
@@ -609,6 +622,23 @@ def startup_walk_for_missed_files(self):
 
                 filetype = FileType.Timeline if wal.TIMELINE_RE.match(filename) else FileType.Wal
 
+                # verify if file was already compressed, otherwise the transfer agent will encounter
+                # duplicated UploadEvents. In case it was compressed, we should just add it to the deletion queue
+                base_compressed_file_path = (
+                    compressed_timeline_path if filetype is FileType.Timeline else compressed_xlog_path
+                )
+                if os.path.isfile(os.path.join(base_compressed_file_path, filename)):
+                    self.log.debug(
+                        "Uncompressed file %r is already compressed, adding to deletion queue.", full_path
+                    )
+                    if filetype is FileType.Timeline:
+                        os.unlink(full_path)
+                    else:
+                        delete_request = WalFileDeletionEvent(backup_site_name=site, file_path=Path(full_path))
+                        self.wal_file_deletion_queue.put(delete_request)
+                        self.log.info("Adding to Uncompressed WAL file to deletion queue: %s", full_path)
```

Comment:
added only WAL file types, since compressor only unlinks timeline types. So, better to have same logic in both places